jax/experimental/compilation_cache/compilation_cache.py

```diff
@@ -14,7 +14,44 @@
 
 import hashlib
 import jax
+from jax.experimental.compilation_cache.file_system_cache import FileSystemCache
+from jax.lib import xla_client
+from typing import Optional
+import warnings
 
+cache = None
+
+def initialize_cache(path):
+    """Creates a global cache object. Should only be called once per process."""
+    global cache
+    cache = FileSystemCache(path)
+
+def get_executable(xla_computation, compile_options) -> Optional[xla_client.Executable]:
+    """Returns the cached executable if present, or None otherwise."""
+    if not cache:
+        return None
+    cache_key = get_cache_key(xla_computation, compile_options)
+    xla_executable_serialized = cache.get(cache_key)
+    if not xla_executable_serialized:
+        return None
+    backend = jax.lib.xla_bridge.get_backend()
+    #TODO(skye): xla_computation.get_hlo_module() is the unoptimized HLO but it should
+    #be optimized
+    xla_executable_deserialized = backend.deserialize_executable(
+                                  xla_executable_serialized,
+                                  xla_computation.get_hlo_module(),
+                                  compile_options)
+    return xla_executable_deserialized
+
+def put_executable(xla_computation, compile_options, executable: xla_client.Executable):
+    """Adds 'executable' to the cache, possibly evicting older entries."""
+    if not cache:
+        warnings.warn("file path for cache has not been specified")
+        return
```

Comment:
Same here, make this `assert cache is not None`, and add a note to the docstring.